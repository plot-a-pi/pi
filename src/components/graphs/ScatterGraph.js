import React, { useRef, useEffect, useState } from 'react';
import { PropTypes } from 'prop-types';
import styles from './ScatterGraph.css';
import { createD3Scatterplot } from '../../d3/scatterplot';

const ScatterGraph = ({ data, xMax, yMax, xLabel, yLabel, title }) => {
  const [containerWidth, setContainerWidth] = useState(800);
  const [containerHeight, setContainerHeight] = useState(800);

  const d3Container = useRef(null);
  useEffect(
    () => {
      if(data && d3Container.current) {
        createD3Scatterplot(d3Container, data, xMax, yMax, xLabel, yLabel, title, containerWidth, containerHeight);
      }
    },
    [data, d3Container.current]);
  return (
    <svg
      className="d3-component"

      // these need to adjust based on media queries and page dependent:  home or big graph
      width={containerWidth}
      height={containerHeight}
      ref={d3Container}
    />
  );
};

ScatterGraph.propTypes = {
  data: PropTypes.array.isRequired,
  xMax: PropTypes.number.isRequired,
  yMax: PropTypes.number.isRequired
};

export default ScatterGraph;
