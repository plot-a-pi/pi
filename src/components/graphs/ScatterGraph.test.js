import React from 'react';
import { shallow } from 'enzyme';
import ScatterGraph from './ScatterGraph';

describe('ScatterGraph component', () => {
  it('renders ScatterGraph', () => {
    const wrapper = shallow(<ScatterGraph />);
    expect(wrapper).toMatchSnapshot();
  });
});
