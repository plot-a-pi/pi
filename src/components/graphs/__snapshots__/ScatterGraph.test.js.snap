// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScatterGraph component renders ScatterGraph 1`] = `
<svg
  className="d3-component"
  height={800}
  width={800}
/>
`;
